packages/video_player/video_player_web/lib/video_player_web.dart

```diff
@@ -66,21 +66,22 @@ class VideoPlayerPlugin extends VideoPlayerPlatform {
     final int textureId = _textureCounter;
     _textureCounter++;
 
-    String uri;
+    late String uri;
     switch (dataSource.sourceType) {
       case DataSourceType.network:
         // Do NOT modify the incoming uri, it can be a Blob, and Safari doesn't
         // like blobs that have changed.
         uri = dataSource.uri;
         break;
       case DataSourceType.asset:
-        String assetUrl = dataSource.asset;
-        if (dataSource.package != null && dataSource.package.isNotEmpty) {
+        String assetUrl = dataSource.asset!;
+        if (dataSource.package != null && dataSource.package!.isNotEmpty) {
           assetUrl = 'packages/${dataSource.package}/$assetUrl';
         }
         // 'webOnlyAssetManager' is only in the web version of dart:ui
         // ignore: undefined_prefixed_name
-        assetUrl = ui.webOnlyAssetManager.getAssetUrl(assetUrl);
+        ui.webOnlyAssetManager.getAssetUrl(assetUrl);
+        assetUrl = '';
```

Comment:
Not sure why we need this line, could you elaborate?

Reply:
my mistake. Fixed.